web/src/components/Accordion.tsx

```diff
@@ -34,18 +39,45 @@ export default function Accordion({
   setState: (isCollapsed: boolean) => void;
   isTopExpanding?: boolean;
 }) {
+  const { ref, height: observerHeight } = useResizeObserver<HTMLDivElement>();
+  const [renderChildren, setRenderChildren] = useState(!isCollapsed);
+
+  const [spring, api] = useSpring(
+    () => ({
+      height: isCollapsed ? 0 : observerHeight,
+      // eslint-disable-next-line unicorn/no-nested-ternary -- it interferes with prettier
+      rotate: isCollapsed ? (isTopExpanding ? -90 : 90) : isTopExpanding ? 90 : -90,
+    }),
+    [isCollapsed, isTopExpanding, observerHeight]
+  );
+
   const handleToggleCollapse = () => {
     onClick?.();
 
     isCollapsed && onOpen?.();
 
+    setRenderChildren(true);
```

Comment:
I know this might be a bit old school, but consider refactoring `handleToggleCollapse` to use if/else instead (see snippet below).
1. I think it's a little easier to read the intent. One can see the open and the close cases clearly.
2. We don't have to rely on a closure, which can cause confusion. At first glance, `onRest: () => setRenderChildren(isCollapsed)` looks like we are intending to `setRenderChildren` to true when we're collapsed and false when we're open, which is the opposite behavior of what we would expect. It works because the onRest closure captured the `isCollapsed` at the time that the callback was created, which is the previous state so it will work as expected, but it's hard to understand immediately.
3. We don't need to toggle the eslint rule.

```javascript
  const handleToggleCollapse = () => {
    onClick?.();
    setRenderChildren(true);

    if (isCollapsed) {
      onOpen?.();

      api.start({
        to: {
          height: observerHeight,
          rotate: isTopExpanding ? 90 : -90,
        }
      });
    } else {
      api.start({
        to: {
          height: 0,
          rotate: isTopExpanding ? -90 : 90,
        },
        onRest: () => {
          setRenderChildren(false);
        },
      });
    }

    setState(!isCollapsed);
  };
  ```

Reply:
Tried it locally and it seems to work just as well and is a lot more readable!

Feel free to commit it directly or add it as a suggestion instad so you get the credit that you very much deserve!

PS: I like to take size and performance optimisations to the extreme, maybe a bit to far at time so don't hesitate to call me out on it if it goes a step to far 😅

Reply:
Updated. And it's good to have many different perspectives!